### DISCUSSION TITLE
Close all Modals

### DISCUSSION THREAD

**Discussion by bgriffy** [Megabit/Blazorise#3524] [Q&A]
Is there any possible way to close all open blazorise modals in my application?

**Comment by stsrki** [comment]
Well, as long as you hold the reference to all of them it is possible. Simply `modalRef.Hide()` would work. But I have a question. Why do you have more than one modal open in the app?

**Reply by bgriffy** [reply]
I guess we're really not going to have more than one modal open at a time. But I'm mainly just trying to find a way to close any modal that is open without having a direct reference to it. The scenario I'm running into is that anytime an error is produced anywhere the app, I want to show a popup modal with the error message and a button to refresh. Example: 

- On Index page, there is an IndexGrid component 
- Double clicking a row in the grid brings up IndexDetail modal component 
- An error occurs on the Index page while the IndexDetail modal is open 
- We show a new modal popup modal to the user, displaying the error message 

Since the index page doesn't have a direct ref to that IndexDetail modal, we can't close it from here. So the error message popup won't show up because the IndexDetail is overlaying it. 

 I was thinking maybe there is some logic on the JS side I could call to close all instances of open blazorise modals?

**Reply by stsrki** [reply]
If you only need to show the error message you can use our message or notification service.

- https://blazorise.com/docs/services/message
- https://blazorise.com/docs/services/notification

**Reply by bgriffy** [reply]
That message service would work, but I need to have a "Refresh" button on it.

**Reply by stsrki** [reply]
You can set the OK button text through message service options and define it as "Refresh". But I'm not sure what you want to do with Refresh and what does it refresh? It reverts the page to it's original state?

**Reply by bgriffy** [reply]
The refresh button is just to reload the app. The same as the default "Unhandled Exception" banner that pops up in blazor. 

Btw, I looked into writing some JS to just close all open modals and came up with this: 

``` 
function closeAllModals() {
    // Get main content section 
    const content = document.querySelector(".content");

    // Get modals
    const modals = document.getElementsByClassName('modal');

    for (let i = 0; i < modals.length; i++) {
        modals[i].classList.remove('show');
        modals[i].setAttribute('aria-hidden', 'true');
        modals[i].setAttribute('style', 'display: none');
    }

    // Get modal backdrops
    const modalsBackdrops = document.getElementsByClassName('modal-backdrop');

    // Remove every modal backdrop
    for (let i = 0; i < modalsBackdrops.length; i++) {
        content.removeChild(modalsBackdrops[i]);
    }
}
``` 
Can you think of any potential issues with doing this?

**Reply by stsrki** [reply]
That would remove classes and other attributes from the DOM. But the internal state of the Modal component would still be the same and could lead to sidefects.
